src/AppBundle/Subscriber/StatusChangeByReviewSubscriber.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace AppBundle\Subscriber;
+
+use AppBundle\Event\GitHubEvent;
+use AppBundle\GitHubEvents;
+use AppBundle\Issues\Status;
+use AppBundle\Issues\StatusApi;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Changes the status when a new review is submitted.
+ *
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ */
+class StatusChangeByReviewSubscriber extends AbstractStatusChangeSubscriber
+{
+    private $logger;
+
+    public function __construct(StatusApi $statusApi, LoggerInterface $logger)
+    {
+        parent::__construct($statusApi);
+        $this->logger = $logger;
+    }
+
+    /**
+     * Sets the status based on the review state (approved/changes requested) 
+     * or the review body (using the Status: keyword).
+     *
+     * @param GithubEvent $event
+     */
+    public function onReview(GitHubEvent $event)
+    {
+        $data = $event->getData();
+        if ('submitted' !== $data['action']) {
+            $event->setResponseData(array('unsupported_action' => $data['action']));
+
+            return;
+        }
+
+        $repository = $event->getRepository();
+        $pullRequestNumber = $data['pull_request']['number'];
+        $newStatus = null;
+
+        // Set status based on review state
+        switch (strtolower($data['review']['state'])) {
+            case 'approved':
+                $newStatus = Status::REVIEWED;
+
+                break;
+            case 'changes_requested':
+                $newStatus = Status::NEEDS_WORK;
+
+                break;
+            default:
+                $newStatus = $this->parseStatusFromText($data['review']['body']);
+
+                if (Status::REVIEWED === $newStatus && false === $this->checkUserIsAllowedToReview($data)) {
+                    $newStatus = null;
+                }
+        }
+
+        $event->setResponseData(array(
```

Comment:
What about moving the beginning of the method and those line in the abstract listener too, it could call an abstract method to extract the status from the payload, wdyt?